core/vnl/algo/vnl_matrix_inverse.h

```diff
@@ -34,7 +34,14 @@ struct vnl_matrix_inverse : public vnl_svd<T>
   vnl_matrix_inverse(vnl_matrix<T> const & M): vnl_svd<T>(M) { }
   ~vnl_matrix_inverse() override = default;
 
-  operator vnl_matrix<T> () const { return this->inverse(); }
+  vnl_matrix<T> as_matrix() const { return this->inverse(); }
+
+#if VXL_LEGACY_FUTURE_REMOVE
+  explicit operator vnl_matrix<T>() const { return this->inverse(); }
+#else
+  //VXL_DEPRECATED_MSG("Implicit cast conversion is dangerous.\nUSE: .as_vector() or .as_ref() member function for clarity.")
```

Comment:
Why is this commented out?

Reply:
Fixed and improved upon to allow selection of how conversions should occur.  Needed to add a separate option for 

```
# Historically VXL implicitly converts between vnl_vector_fixed and vnl_vector or vnl_vector_ref ( & matrix types too),
# when VXL_USE_HISTORICAL_IMPLICIT_CONVERSIONS:BOOL=OFF, require explicit conversions to occur, this simplifies finding
# performance eating conversions by having the compiler throw errors until an explicit conversion is put into place.
option(VXL_USE_HISTORICAL_IMPLICIT_CONVERSIONS "Set backward compatible implicit conversion of types without warning or error when ON" ON)
mark_as_advanced(VXL_USE_HISTORICAL_IMPLICIT_CONVERSIONS)
```